tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php

```diff
@@ -316,6 +316,26 @@ public function testVariables(): void
                 3,
                 'differentRenderConditional.latte',
             ],
+            [
+                'Dumped type: mixed', // TODO should be string, but PHPStan resolves this as mixed for now
```

Comment:
https://phpstan.org/r/c5055b61-4862-4f5b-bfb8-7d570e68fb73

Reply:
After discussion with @ondrejmirtes, issue has been created https://github.com/phpstan/phpstan/issues/9331, maybe there is a light at the end of the tunnel

Reply:
This could be source of unexpected sideeffects. I would wait for solving it in PHPStan before merging this.

Reply:
Well yes, with mixed type it could be. On the other hand, the most questions I get are about “variable is always defined” error 😂

Reply:
Well at this point it is choosing between two non-ideal options. But loosing type of all possibly undefined variables is IMHO worse. Maybe ignore "always defined" errors as temporary solution?

Reply:
I would like to not ignore errors at all. This error is fixable with adding variable to each branch of program. I'll mark this as draft while there will be some support from phpstan itself

Reply:
I think I solved this with few extract calls for each type of variables (variables, default variables, filters, functions) instead of `@var` annotations. Everything works as expected in tests, need some testing in real world apps. Maybe you could check this branch too

Reply:
I will check next week. I will be without computer for rest of the week.

Reply:
I've removed transformer and changed error messages. If you don't have time, I'd merge it

Reply:
I am back home but with bad cough and on top of that eye infection :-( So I do not feel for doing reviews. if tests pass then you can merge it.